### PR TITLE
Harden endpoint signaling and identity handling

### PR DESCRIPTION
**Pull request by HashimTheArab**
## Summary

- add timeouts to the ServeTLS HTTP server and keep normal shutdown from recording http.ErrServerClosed as the handler context cause
- bound SDP request and response body reads, and stop logging full SDP offers on negotiation failures
- replace identity error code 37 with ErrorCodeIdentityVerificationFailed
- make endpoint handler tests avoid the package-global notify check toggle
- add focused identity tests for sign/verify, tampered fingerprints, expired tokens, and nil cpk marshal errors

## Notes

This intentionally does not redesign endpoint.Handler's Signaling semantics or change the identity verification trust model. Those are broader API/design decisions and should stay separate from this hardening cleanup.

The SDP body limit is 1 MiB to avoid clipping identity-bearing or bundled-candidate SDP while still bounding untrusted HTTP input.

## Validation

- go test . -run 'TestIdentity|TestClaimPublicKey|TestTokenClaims' -count=1
- go test ./endpoint -run 'TestClientSignal|TestHandler|TestServeTLS' -count=1
- go test ./...
- go vet ./...
- go test -race ./endpoint
- git diff --check

### CODE REVIEW

File: endpoint/handler.go [hunk 1]
```diff
@@ -277,7 +307,12 @@ func (h *Handler) handleOffer(w http.ResponseWriter, req *http.Request) {
 
 	signal, err := h.negotiate(ctx, networkID, string(b))
 	if err != nil {
-		log.Error("error negotiating", slog.String("offer", string(b)), slog.Any("error", err))
+		offerHash := sha256.Sum256(b)
```
Comment (lactyy2) [inline endpoint/handler.go]: I don't think a hash computed from the SDP offer is particularly useful info. could you simply log the offer itself?

**Review by lactyy2 — CHANGES_REQUESTED** [review]

File: endpoint/handler.go (hunk 1 above)
Comment (HashimTheArab) [reply endpoint/handler.go]: done, thanks
